salt/pillar/git_pillar.py

```diff
@@ -415,6 +415,7 @@ def ext_pillar(minion_id, pillar, *repos):  # pylint: disable=unused-argument
         # If masterless, fetch the remotes. We'll need to remove this once
         # we make the minion daemon able to run standalone.
         git_pillar.fetch_remotes()
+    git_pillar.global_lock()
```

Comment:
I did just notice that you're not using a `with` block, which you need since you're using `contextlib.contextmanager`.